var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -281,21 +283,45 @@ def blas_lapack_pkg_config(self) -> Tuple[str, str]:
 
         return blas, lapack
 
-    @when("@1.26:")
+    @when("@1.25:")
     def config_settings(self, spec, prefix):
+        settings = {"builddir": "build", "compile-args": f"-j{make_jobs}"}
+
+        if self.spec.satisfies("@1.26:"):
+            settings.update(self.blas_config_settings())
+
+        # Disable AVX512 features for Intel Classic compilers
+        # https://numpy.org/doc/stable/reference/simd/build-options.html
+        # https://github.com/numpy/numpy/issues/27840
+        # https://github.com/matplotlib/matplotlib/issues/28762
+        archs = ("x86_64_v4:", "cannonlake:", "mic_knl")
+        if any([self.spec.satisfies(f"target={arch} %intel") for arch in archs]):
+            intel_settings = {
+                "setup-args": {
+                    "-Dcpu-dispatch": (
+                        settings.get("setup-args", {}).get("-Dcpu-dispatch", "")
+                        + " "
+                        + "MAX -AVX512F -AVX512CD -AVX512_KNL -AVX512_KNM -AVX512_SKX "
+                        + "-AVX512_CLX -AVX512_CNL -AVX512_ICL -AVX512_SPR"
+                    )
+                }
+            }
+            settings.update(intel_settings)
+
+        return settings
+
+    def blas_config_settings(self):
         blas, lapack = self.blas_lapack_pkg_config()
         return {
-            "builddir": "build",
-            "compile-args": f"-j{make_jobs}",
             "setup-args": {
                 # https://scipy.github.io/devdocs/building/blas_lapack.html
                 "-Dblas": blas,
                 "-Dlapack": lapack,
                 # https://numpy.org/doc/stable/reference/simd/build-options.html
                 # TODO: get this working in CI
                 # "-Dcpu-baseline": "native",
-                # "-Dcpu-dispatch": "none",
-            },
+                # "-Dcpu-dispatch": "none ",
```

Comment:
Is this whitepace needed?

Reply:
No, it isn't.
```suggestion
                # "-Dcpu-dispatch": "none",
```

Reply:
Hmm, for some reason the whitespace still appears to be there.  I'll try to remove it on console.

Reply:
Just needed to refresh...